native/src/modules/endpoint/sagas/fetchResourceCache.js

```diff
@@ -32,12 +52,18 @@ export default function * fetchResourceCache (
       return acc
     }, {})
 
-    const results = yield call(new FetcherModule().fetchAsync, targetFilePaths, progress => console.log(progress))
+    if (FetcherModule.currentlyFetching) {
+      throw new Error('Already fetching!')
+    }
```

Comment:
Why is this necessary here?

Reply:
I could not throw the error in FetcherModule.createProgressChannel itself because I'm forking a new process here.
Throwing an error in createProgressChannel itself did not cancel this workflow here.
 
So I decided to check it before creating the progress channel here. 

Perhaps you have a suggestion how to do it inside of createProgressChannel itself, it would be nicer.

Reply:
Why do we have to throw an error at all in createProgressChannel if it is already fetching? Isn't this already handled in `fetchAsync`?

Reply:
Ah I see, you removed it in fetchAsync. Would it be a problem if a channel is created while a fetch is in progress?

Reply:
... I'm not sure^^
If an error happens in the "FetcherModule().fetchAsync" call then "yield cancel(progressTask)" would not be called.
But altogether is surrounded by a try catch, does the forked task also gets closed also if the try fails?

Reply:
I will leave it this way, just to be sure